File: fast_sync_dart_client/fast_sync_core/lib/src/implementation/metadata/syncable_object.dart
import 'package:fast_sync_client/src/absraction/metadata/isync_metadata.dart';
import 'package:fast_sync_client/src/absraction/metadata/isync_operation.dart';
import 'package:fast_sync_client/src/absraction/metadata/isyncalbe_object.dart';

class SyncableObject implements ISyncableObject {
  @override
  final String id;

  @override
  final ISyncMetadata metadata;

  @override
  final bool deleted;

  final SyncOperationEnum syncOperation;

  SyncableObject(
      {required this.id,
      required this.metadata,
      required this.deleted,
      required this.syncOperation});

  @override
  num getVersion() {
    return metadata.version;
  }
}

File: fast_sync_typescript_service_sdk/packages/fast-sync-core/src/core/implementation/metadata/syncable_object.ts
import { ISyncMetadata } from "../../abstraction/metadata/ISync_metadata";
import { SyncOperationEnum } from "../../abstraction/metadata/ISync_operation";
import { ISyncableObject } from "../../abstraction/metadata/ISyncable_object";

export class SyncableObject implements ISyncableObject {
    public id: string;
    public metadata: ISyncMetadata;
    public deleted: boolean;
    syncOperation: SyncOperationEnum;
    constructor() {}
    
    getVersion(){
      return this.metadata.version
    }
  
    setVersion(version: number): number{
      return this.metadata.version = version;
    } 
  }